x-pack/plugin/esql/qa/server/mixed-cluster/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/mixed/Clusters.java

```diff
@@ -14,15 +14,21 @@
 public class Clusters {
     public static ElasticsearchCluster mixedVersionCluster() {
         Version oldVersion = Version.fromString(System.getProperty("tests.old_cluster_version"));
-        return ElasticsearchCluster.local()
+
+        var cluster = ElasticsearchCluster.local()
             .distribution(DistributionType.DEFAULT)
             .withNode(node -> node.version(oldVersion))
             .withNode(node -> node.version(Version.CURRENT))
             .withNode(node -> node.version(oldVersion))
             .withNode(node -> node.version(Version.CURRENT))
             .setting("xpack.security.enabled", "false")
             .setting("xpack.license.self_generated.type", "trial")
-            .setting("cluster.routing.rebalance.enable", "none") // disable relocation until we have retry in ESQL
-            .build();
+            .setting("cluster.routing.rebalance.enable", "none"); // disable relocation until we have retry in ESQL
+
+        // The plugin cannot be loaded on mixed version clusters with versions on 8.x
+        if (oldVersion.onOrAfter("9.0.0")) {
+            cluster = cluster.plugin("inference-service-test");
```

Comment:
For my one knowledge, why can the service not be loaded on ES 8.x ? Is this a short-term limitation or something else?

Reply:
It is a short-term limitation until we fix the plugin

Reply:
the only consequence for now is that we won't execute the test from `x-pack/plugin/esql/qa/testFixtures/src/main/resources/semantic_text.csv-spec` for mixed-version and multi-cluster

we do execute it for multi-node and single-node.
This only seems to happen when we try to set the plugin for clusters with 8.x nodes.

Reply:
Ok, got it. This workaround for the temporary limitation seems reasonable. Is there an issue tracking the plugin issue?    

Also, just to confirm. Does the test plugin work ok on an 8.x only cluster - not mixed, just 8.x nodes. For example, if this change were to be backported to the Elasticsearch 8.x branch.

Reply:
We have another plugin `test-infererence-plugin` which is used to test retrievers for which we don't have any mixed version clusters or multi cluster tests.
Because of this I think it is fine we figure out how to enable these tests as a follow up and potentially find a solution we can use not just for ES|QL integration tests.